src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -869,6 +864,36 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
             return retVal;
         }
 
+        /// <summary>
+        /// Prepare headers in a format that can be fed to curl API
+        /// </summary>
+        /// <param name="headers"></param>
+        /// <returns>List of headers</returns>
+        private static List<string> PrepareRequestHeaders(params HttpHeaders[] headers)
+        {
+            List<string> allHeaders = new List<string>(headers.Length);
+
+            foreach (var multiHeaders in headers)
+            {
+                if (multiHeaders != null)
+                {
+                    foreach (var header in multiHeaders)
+                    {
+                        string headerStr = header.Key + ":";
+
+                        foreach (var headerValue in header.Value)
+                        {
+                            headerStr += " " + headerValue;
```

Comment:
What does libcurl do with space separated values? Is there an doc reference for this?

Multi value headers should be comma separated or sent as multiple headers. E.g.:
```
Cookie: name=value
Cookie: name1=value1
```
Some applications can't handle the comma separated format, so it's best to use the multiple headers format.